boards/DCM/Src/App/states/App_DriveState.c

```diff
@@ -9,30 +9,54 @@
-    const float right_motor_speed_rpm =
-        (float)abs(App_CanRx_INVR_MOTOR_POSITION_INFO_GetSignal_D2_MOTOR_SPEED_INVR(can_rx));
-    const float left_motor_speed_rpm =
-        (float)abs(App_CanRx_INVL_MOTOR_POSITION_INFO_GetSignal_D2_MOTOR_SPEED_INVL(can_rx));
-    float bms_torque_limit = MAX_TORQUE_REQUEST_NM;
-    float fsm_torque_limit = App_CanRx_FSM_TORQUE_LIMITING_GetSignal_FSM_TORQUE_LIMIT(can_rx);
-
-    if ((right_motor_speed_rpm + left_motor_speed_rpm) > 0.0f)
+    // Set the latch high when the brake is pressed. Clear the latch when the brake is not actuated and the APPS returns
+    // to a position that is lower than 5%
+    static bool brake_actuated_latch = false;
+
+    // Acquire the motor RPM from the FSM via CAN
+    const float right_motor_speed_rad_s =
+        RPM_TO_RADS((float)abs(App_CanRx_INVR_MOTOR_POSITION_INFO_GetSignal_D2_MOTOR_SPEED_INVR(can_rx)));
+    const float left_motor_speed_rad_s =
+        RPM_TO_RADS((float)abs(App_CanRx_INVL_MOTOR_POSITION_INFO_GetSignal_D2_MOTOR_SPEED_INVL(can_rx)));
+
+    // Calculate the maximum torque request to draw the maximum power available from the BMS
+    float calculated_torque_limit = MAX_TORQUE_REQUEST_NM;
+    if ((right_motor_speed_rad_s + left_motor_speed_rad_s) > 0.0f)
     {
-        // Estimate the maximum torque request to draw the maximum power available from the BMS
-        bms_torque_limit = bms_available_power * EFFICIENCY_ESTIMATE /
-                           (RPM_TO_RADS(right_motor_speed_rpm) + RPM_TO_RADS(left_motor_speed_rpm));
+        calculated_torque_limit = App_CanRx_BMS_AVAILABLE_POWER_GetSignal_AVAILABLE_POWER(can_rx) *
+                                  EFFICIENCY_ESTIMATE / (right_motor_speed_rad_s + left_motor_speed_rad_s);
     }
 
-    // Calculate the maximum torque request to scale pedal percentage off of
-    const float max_torque_request = min(bms_torque_limit, MAX_TORQUE_REQUEST_NM);
+    const float apps_pct = App_CanRx_FSM_PEDAL_POSITION_GetSignal_MAPPED_PEDAL_PERCENTAGE(can_rx);
+    if (App_CanRx_FSM_BRAKE_GetSignal_BRAKE_IS_ACTUATED(can_rx) && (apps_pct >= BRAKE_PLAUSIBILITY_LATCH_SET_THRESH))
+    {
+        // Set the brake actuated latch high only when the brake is actuated
+        brake_actuated_latch = true;
+    }
+    else if (brake_actuated_latch && (apps_pct < BRAKE_PLAUSIBILITY_LATCH_RESET_THRESH))
+    {
+        brake_actuated_latch = false;
+    }
 
-    // Calculate the actual torque request to transmit
-    const float torque_request =
-        min(0.01f * App_CanRx_FSM_PEDAL_POSITION_GetSignal_MAPPED_PEDAL_PERCENTAGE(can_rx) * max_torque_request,
-            fsm_torque_limit);
+    // Calculate the torque request to transmit to the inverter
+    float torque_request = 0.0f;
+    if (!brake_actuated_latch)
+    {
+        const float max_torque_request = min(calculated_torque_limit, MAX_TORQUE_REQUEST_NM);
```

Comment:
what happens if motor speeds are zero? calculated_torque_limit would be the same as MAX_TORQUE_REQUEST_NM. I want to confirm that this will not break our min function.

Reply:
No, it will not since the min function is simply a macro that we have defined in `App_SharedMacros.h`: 

 `#define min(a,b) ((a) < (b) ? (a) : (b))`